src/systems-swamp/stdio.py

```diff
@@ -0,0 +1,58 @@
+import builtins
+import contextlib
+import sys
+from io import StringIO
+from textwrap import indent
+from typing import Any, TextIO, TypeVar
+
+T = TypeVar("T", bound=TextIO)
+
+SAVED_INPUT = input
+
+
+def _input(prompt: Any) -> str:
+    result = SAVED_INPUT(prompt)
+    print(result)
+    return result
+
+
+@contextlib.contextmanager
+def redirect_stdin(target: TextIO, write_to_stdout: bool = False):
```

Comment:
```suggestion
def redirect_stdin(target: TextIO, *, write_to_stdout: bool = False) -> Iterator[None]:
```
\+ `from collections.abc import Iterator` please

Reply:
Added in commit